server/live_query/redis_live_query.go

```diff
@@ -316,14 +320,92 @@ func (r *redisLiveQuery) removeQueryNames(names ...string) error {
 }
 
 func (r *redisLiveQuery) LoadActiveQueryNames() ([]string, error) {
+	r.cache.mu.RLock()
+	if r.cache.cacheExp.After(time.Now()) {
+		names := r.cache.activeQueriesCache
+		r.cache.mu.RUnlock()
+		return names, nil
+	}
+	r.cache.mu.RUnlock()
+
+	cacheCopy, err := r.loadCache()
+	if err != nil {
+		return nil, fmt.Errorf("load cache: %w", err)
+	}
+
+	return cacheCopy.activeQueriesCache, nil
+}
+
+func (r *redisLiveQuery) loadCache() (memCache, error) {
+	expiredQueries := make(map[string]struct{})
+	sqlCache := make(map[string]string)
 	conn := redis.ConfigureDoer(r.pool, r.pool.Get())
 	defer conn.Close()
 
-	names, err := redigo.Strings(conn.Do("SMEMBERS", activeQueriesKey))
-	if err != nil && err != redigo.ErrNil {
-		return nil, err
+	activeIDs, err := redigo.Strings(conn.Do("SMEMBERS", activeQueriesKey))
+	if err != nil {
+		return memCache{}, fmt.Errorf("get active queries: %w", err)
 	}
-	return names, nil
+
+	for _, id := range activeIDs {
```

Comment:
The thought is to optimize this for the zero-trust workflow, which is expecting about 25 live query requests per second, each query scoped to a single host.  caching on demand wouldn't be too useful because the cache would be stored and then never used again.  If we instead cache 25+ queries for 1sec, then 24+ hosts would use the cached sql data.

In reality we'll have more than 25 queries cached because of the default 10s distributed interval and query execution time, but that's the general concept.